lib/wasmex/component_store.ex

```diff
@@ -0,0 +1,26 @@
+defmodule Wasmex.ComponentStore do
+  alias Wasmex.Engine
+  alias Wasmex.Wasi.WasiOptions
+
+  defstruct resource: nil, reference: nil
+
+  def __wrap_resource__(resource) do
+    %__MODULE__{
+      resource: resource,
+      reference: make_ref()
+    }
+  end
+
+  def new(%WasiOptions{} = options, store_limits \\ nil, engine \\ nil) do
+    %Engine{resource: engine_resource} = engine || Engine.default()
+
+    case Wasmex.Native.store_new_wasi_p2(
```

Comment:
hmm, do we want to separate wasi and non-wasi components maybe? or do all wasm components need to have wasi enabled?

Reply:
I refactored this to have a WASIP2Options. I *think* all components need some amount of WASI enabled, but the options are different. Lemme know if you think this approach is good

Reply:
hmm, not sure. I need to research better but my understanding so far is that wasi-p2 is just a couple of different worlds a component can depend on and wasmtime gives a default implementation if we enable it to do so. But we could totally tell wasmtime to NOT enable any WASI P2 and instead implement the world in Elixir.

I want to have the flexibility to do that and have a default which is 100% sandboxes. There is some security and usability background to it. I want users of wasmex to explicitly and knowingly enable API surface to their components. Nobody should be surprised that an attacker could suddenly read from stdin, do HTTP calls, get the current time or anything else potentially security sensitive operation.

Reply:
That makes sense, I very much agree we don't want to give the user a foot gun. I'll do some experiments and see if things still work with WASI disabled. If not we can talk about what to do.